File: Components/WinForms/VB/PivotGridMainDemo/Modules/RunningTotal.vb
Imports Microsoft.VisualBasic
Imports System
Namespace DevExpress.XtraPivotGrid.Demos.Modules
	Partial Public Class RunningTotal
		Inherits TutorialControl
		Public Sub New()
			CreateWaitDialog()
			InitializeComponent()

            TutorialInfo.WhatsThisCodeFile = "VB\PivotGridMainDemo\Modules\RunningTotal.vb"
            TutorialInfo.WhatsThisXMLFile = "RunningTotal.xml"

			pivotGridControl.DataSource = GetNWindData("CustomerReports")
			ceShowRunningTotals.Checked = True
			ceAllowCrossGroupRunningTotals.Checked = True
			pivotGridControl.OptionsCustomization.AllowFilterBySummary = False
		End Sub
		Public Overrides ReadOnly Property ViewOptionsControl() As PivotGridControl
			Get
				Return pivotGridControl
			End Get
		End Property
		Public Overrides ReadOnly Property ExportControl() As PivotGridControl
			Get
				Return pivotGridControl
			End Get
		End Property
		'<ceShowRunningTotals>
		Private Sub ceShowRunningTotals_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceShowRunningTotals.CheckedChanged
			fieldOrderQuarter.RunningTotal = ceShowRunningTotals.Checked
			ceAllowCrossGroupRunningTotals.Enabled = ceShowRunningTotals.Checked
		End Sub
		'</ceShowRunningTotals>

		'<ceAllowCrossGroupRunningTotals>
		Private Sub ceAllowCrossGroupRunningTotals_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceAllowCrossGroupRunningTotals.CheckedChanged
			pivotGridControl.OptionsData.AllowCrossGroupVariation = ceAllowCrossGroupRunningTotals.Checked
		End Sub
		'</ceAllowCrossGroupRunningTotals>
	End Class
End Namespace

File: Components/WinForms/CS/PivotGridMainDemo/Modules/RunningTotal.cs

namespace DevExpress.XtraPivotGrid.Demos.Modules {
	public partial class RunningTotal : TutorialControl {
		public RunningTotal() {
			CreateWaitDialog();
			InitializeComponent();

            TutorialInfo.WhatsThisCodeFile = "CS\\PivotGridMainDemo\\Modules\\RunningTotal.cs";
            TutorialInfo.WhatsThisXMLFile = "DevExpress.XtraPivotGrid.Demos.CodeInfo.RunningTotal.xml";

			pivotGridControl.DataSource = GetNWindData("CustomerReports");
			ceShowRunningTotals.Checked = true;
            ceAllowCrossGroupRunningTotals.Checked = true;
            pivotGridControl.OptionsCustomization.AllowFilterBySummary = false;
		}
		public override PivotGridControl ViewOptionsControl { get { return pivotGridControl; } }
		public override PivotGridControl ExportControl { get { return pivotGridControl; } }
        //<ceShowRunningTotals>
		void ceShowRunningTotals_CheckedChanged(object sender, System.EventArgs e) {
			fieldOrderQuarter.RunningTotal = ceShowRunningTotals.Checked;
            ceAllowCrossGroupRunningTotals.Enabled = ceShowRunningTotals.Checked;
		}
        //</ceShowRunningTotals>

        //<ceAllowCrossGroupRunningTotals>
        void ceAllowCrossGroupRunningTotals_CheckedChanged(object sender, System.EventArgs e) {
            pivotGridControl.OptionsData.AllowCrossGroupVariation = ceAllowCrossGroupRunningTotals.Checked;
        }
        //</ceAllowCrossGroupRunningTotals>
	}
}
